scanomatic/ui_server_data/js/som/analysis.js

```diff
@@ -1,15 +1,41 @@
+import $ from 'jquery';
+import {
+  API,
+  Dialogue,
+  getPathSuggestions,
+  getSharedValue,
+  InputEnabled,
+  Map,
+} from './helpers';
+
 let gridplates = null;
 let localFixture = true;
 let path = '';
 
-function analysisToggleLocalFixture(caller) {
-  localFixture = $(caller).prop('checked');
-  InputEnabled($(current_fixture_id), !localFixture);
-  set_fixture_plate_listing();
+function appendRegriddingUI(parent, plateIndex) {
+  parent.append((
+    `<div class='plate-regridding' id='plate-regridding-${plateIndex}' onmouseleave='hideGridImage();'>`
+      + '<fieldset>'
+      + `<img class='grid_icon' src='/images/grid_icon.png' onmouseenter='loadGridImage(${plateIndex - 1});'>`
+      + `<legend>Plate ${plateIndex}</legend>`
+
+      + `<input type='radio' name='plate-regridding-radio-${plateIndex}' value='Keep' checked='checked'>`
+      + `<label id='plate-regridding-keep${plateIndex}'>Keep previous</label><br>`
+
+      + `<input type='radio' name='plate-regridding-radio-${plateIndex}' value='Offset'>`
+      + `<label id='plate-regridding-offset${plateIndex}'>Offset</label>`
+      + `<input type='number' class='plate-offset' id='plate-regridding-offset-d1-${plateIndex}' value='0' name='Offset-d1'>`
+      + `<input type='number' class='plate-offset' id='plate-regridding-offset-d2-${plateIndex}' value='0' name='Offset-d2'><br>`
+
+      + `<input type='radio' name='plate-regridding-radio-${plateIndex}' value='New'>`
+      + `<label id='plate-regridding-new${plateIndex}'>New grid from scratch</label><br>`
+      + '</fieldset>'
+    + '</div>'
```

Comment:
does this really work, these callbacks in text form like `hideGridImage()` in the first line are part of the `som` namespace right so it should be `som.hideGridImage();` and so on.

Also check `<label id='plate-regridding-offset${plateIndex}'>Offset</label>` and `<label id='plate-regridding-new${plateIndex}` if it really should be without a dash before plateIndex as it is in all other cases.

Reply:
Yes, `som.` is needed. Well spotted.

As for the id's I'll try to have a look.

Reply:
Fixed. The ids aren't actually referenced anywhere, so I've added `-`.

Reply:
should be quite searchable with regridding-new and regridding-offset